src/endpointsApi.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const logger = require('./logger')
+const EndpointModel = require('./models/endpoints')
+const {handleServerError, validateEndpoint} = require('./util')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const saveEndPoint = body => {
+  return new Promise((resolve, reject) => {
+    const endpoint = new EndpointModel(body)
+    endpoint.save(err => {
+      if (err) reject(err)
+      resolve(endpoint)
+    })
+  })
+}
+
+const updateEndPoint = (endpointId, body) => {
+  return new Promise((resolve, reject) => {
+    EndpointModel.findOneAndUpdate(
+      {_id: endpointId},
+      body,
+      {new: true, runValidators: true},
+      (err, result) => {
+        if (err) reject(err)
+        resolve(result)
+      }
+    )
+  })
+}
+
+const createCreateEndpoint = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
+
+    try {
+      await KoaBodyParser()(ctx, () => {})
+
```

Comment:
@bradsawadye Maybe you could have a Content-Type Check. If I send though, content type Text or javascript the request just hangs. Same thing happens if I send through an empty body.

This might be pedantic though... @BMartinos ?

Reply:
Do we not do something like this already to verify the incoming content type is the same as specified in the meta.json?
https://github.com/jembi/openhim-mediator-mapping/blob/master/src/middleware/parser.js#L61-L71

Reply:
@MattyJ007  it does not hang. A bad request response is sent out